relay-metrics/src/aggregatorservice.rs

```diff
@@ -128,144 +94,250 @@ pub struct FlushBuckets {
             let bucket_count = project_buckets.len() as u64;
             relay_statsd::metric!(
                 histogram(MetricHistograms::BucketsFlushedPerProject) = bucket_count,
-                aggregator = self.aggregator.name(),
+                aggregator = aggregator_name,
             );
             total_bucket_count += bucket_count;
 
-            let partitioned_buckets = self
-                .aggregator
-                .partition_buckets(project_buckets, self.flush_partitions);
+            let partitioned_buckets = self.partition_buckets(aggregator_name, project_buckets);
+
             for (partition_key, buckets) in partitioned_buckets {
-                self.process_batches(buckets, |batch| {
-                    if let Some(ref receiver) = self.receiver {
-                        receiver.send(FlushBuckets {
-                            project_key,
-                            partition_key,
-                            buckets: batch,
-                        });
-                    }
-                });
+                let capped_batches =
+                    CappedBucketIter::new(buckets.into_iter(), self.max_flush_bytes);
+
+                let num_batches = capped_batches
+                    .map(|batch| {
+                        relay_statsd::metric!(
+                            histogram(MetricHistograms::BucketsPerBatch) = batch.len() as f64,
+                            aggregator = aggregator_name,
+                        );
+                        if let Some(receiver) = self.receiver.as_ref() {
+                            receiver.send(FlushBuckets {
+                                project_key,
+                                partition_key,
+                                buckets: batch,
+                            });
+                        };
+                    })
+                    .count();
+
+                relay_statsd::metric!(
+                    histogram(MetricHistograms::BatchesPerPartition) = num_batches as f64,
+                    aggregator = aggregator_name,
+                );
             }
         }
         relay_statsd::metric!(
             histogram(MetricHistograms::BucketsFlushed) = total_bucket_count,
-            aggregator = self.aggregator.name(),
+            aggregator = aggregator_name,
         );
     }
 
-    fn handle_merge_buckets(&mut self, msg: MergeBuckets) {
-        let MergeBuckets {
-            project_key,
-            buckets,
-        } = msg;
-        self.aggregator
-            .merge_all(project_key, buckets, self.max_total_bucket_bytes);
+    /// Flushes all the aggregators and sends the [`FlushBuckets`] message to the receiver
+    /// in the fire and forget fashion. It is up to the receiver to send the [`MergeBuckets`]
+    /// message back if buckets could not be flushed and we require another re-try.
+    ///
+    /// If `force` is true, flush all buckets unconditionally and do not attempt to merge back.
```

Comment:
Please keep the docs to the team's coding standards: one short line for the description and then you can elaborate more if appropriate.

Reply:
hey i just moved it 😅  
 
But yeah i can change it, I assumed it wasn't to the standard because its a private function